### PR TITLE
Add datapoint 12.001

### PR DESCRIPTION
**Pull request by ferllings**

### CODE REVIEW

File: knx/dpt/types.go [hunk 1]
```diff
@@ -297,3 +297,42 @@ func (d DPT_9004) Unit() string {
...
+type DPT_12001 uint32
+
+func (d DPT_12001) Pack() []byte {
+	if d <= 0 {
+		return packU32(0)
+	} else if d >= 4294967295 {
+		return packU32(4294967295)
+	} else {
+		return packU32(uint32(d))
+	}
```
Comment (vapourismo) [inline knx/dpt/types.go]: `unit32` can't be lower than 0 or greater than 4294967295.

File: knx/dpt/types.go [hunk 2]
```diff
@@ -297,3 +297,42 @@ func (d DPT_9004) Unit() string {
...
+	if err := unpackU32(data, &value); err != nil {
+		return err
+	}
+
+	// Check the value for valid range
+	if value < 0 {
+		return fmt.Errorf("Counter \"%d\" outside range [0, 4294967295]", value)
+	} else if value > 4294967295 {
+		return fmt.Errorf("Counter \"%d\" outside range [0, 4294967295]", value)
+	}
```
Comment (vapourismo) [inline knx/dpt/types.go]: Neither of these cases can ever fire.

File: knx/dpt/types.go [hunk 3]
```diff
@@ -297,3 +297,42 @@ func (d DPT_9004) Unit() string {
...
+
+	*d = DPT_12001(value)
+
+	return nil
+}
```
Comment (vapourismo) [inline knx/dpt/types.go]: In general this functions could just be `return unpack32(data, (*unit32)(d))`.

**Comment by ferllings** [comment]
Done. That makes it a lot easier. Thanks

**Comment by vapourismo** [comment]
Cheers!

**Pull request merged**